### PR TITLE
[procmon] print new memory protection attributes

### PR DESCRIPTION
**Pull request by zodeak**

### CODE REVIEW

File: src/plugins/procmon/procmon.cpp [hunk 1]
```diff
@@ -629,6 +630,47 @@ static event_response_t open_process_hook_cb(drakvuf_t drakvuf, drakvuf_trap_inf
...
+            printf("procmon Time=" FORMAT_TIMEVAL ",PID=%d,PPID=%d,ProcessName=\"%s\","
+                   "Method=%s,ProcessHandle=0x%" PRIx64 ",",
+                   UNPACK_TIMEVAL(info->timestamp), info->proc_data.pid, info->proc_data.ppid, info->proc_data.name,
+                   info->trap->name, process_handle);
+            printf("NewProtectWin32=");
+            print_protection_attributes(new_protect);
+            break;
+
+        default:
+        case OUTPUT_DEFAULT:
```
Comment (tklengyel) [inline src/plugins/procmon/procmon.cpp]: There is also JSON type output now.

File: src/plugins/procmon/winnt.cpp [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+        attributes &= ~attr;
+        ++num;
+        if (num > 1)
+            printf("%c", sep);
+
+        switch (attr)
+        {
+            case PAGE_NOACCESS:
+                printf("PAGE_NOACCESS");
```
Comment (tklengyel) [inline src/plugins/procmon/winnt.cpp]: I don't like how this function prints strings by itself. I would prefer if this function just returned a `const char*` once to the caller and then the caller prints it.

File: src/plugins/procmon/winnt.cpp [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            default:
+            {
+                char tmp[32] = {0};
+                sprintf(tmp, "0x%" PRIx32, attr);
+                result.append(tmp);
+                break;
+            }
+        }
+    }
+    return result;
```
Comment (tklengyel) [inline src/plugins/procmon/winnt.cpp]: Why not return `result.c_str()`?

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/plugins/procmon/winnt.cpp (hunk 3 above)
Comment (zodeak) [reply src/plugins/procmon/winnt.cpp]: string object will be destroyed, so c_str() returns invalid memory pointer

Comment (tklengyel) [reply src/plugins/procmon/winnt.cpp]: I see

**Comment by tklengyel** [comment]
Thanks!

**Pull request merged**